React default generation is not working

##### **Overview of the issue**

Error when trying to generate an App using default settings for React.
```
[ERROR] /home/ippon/repos/testReact/src/main/java/com/mycompany/myapp/config/CacheConfiguration.java:[38,73] cannot find symbol
  symbol:   variable USERS_BY_LOGIN_CACHE
  location: interface com.mycompany.myapp.repository.UserRepository
[ERROR] /home/ippon/repos/testReact/src/main/java/com/mycompany/myapp/config/CacheConfiguration.java:[39,73] cannot find symbol
  symbol:   variable USERS_BY_EMAIL_CACHE
  location: interface com.mycompany.myapp.repository.UserRepository

```

##### **Motivation for or Use Case**



##### **Reproduce the error**

Just build any React project with default settings.

##### **Related issues**



##### **Suggest a Fix**

Probably have something to do with #7254

##### **JHipster Version(s)**

4.14.0

##### **JHipster configuration**

Using JHipster version installed globally
Executing jhipster:info
Options: 
##### **JHipster Version(s)**

```
test-react@0.0.0 /home/ippon/repos/testReact
└── (empty)

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.14.0",
    "baseName": "testReact",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "react",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>




</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

git version 2.7.4

node: v8.10.0

npm: 5.6.0

yeoman: 2.0.1

yarn: 1.3.2

Docker version 17.12.1-ce, build 7390fc6

docker-compose version 1.16.1, build 6d1ac21

Congratulations, JHipster execution is complete!


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

none

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs
+++ b/generators/server/templates/src/main/java/package/config/CacheConfiguration.java.ejs
@@ -134,7 +134,7 @@ public class CacheConfiguration {
             <%_ if (authenticationType === 'oauth2' && applicationType === 'microservice') { _%>
             cm.createCache("oAuth2Authentication", jcacheConfiguration);
             <%_ } _%>
-            <%_ if (!skipUserManagement || (authenticationType === 'oauth2' && applicationType === 'monolith')) { _%>
+            <%_ if (!skipUserManagement || authenticationType === 'oauth2') { _%>
             cm.createCache(<%=packageName%>.repository.UserRepository.USERS_BY_LOGIN_CACHE, jcacheConfiguration);
             cm.createCache(<%=packageName%>.repository.UserRepository.USERS_BY_EMAIL_CACHE, jcacheConfiguration);
                 <%_ if (enableHibernateCache) { _%>

--- a/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
+++ b/generators/server/templates/src/main/java/package/repository/UserRepository.java.ejs
@@ -82,7 +82,7 @@ import static <%=packageName%>.config.Constants.ID_DELIMITER;
 <%_ if (databaseType === 'sql' || databaseType === 'mongodb' || databaseType === 'couchbase') { _%>
 @Repository
 public interface UserRepository extends <% if (databaseType === 'sql') { %>JpaRepository<User, <% if (authenticationType === 'oauth2') { %>String<% } else { %>Long<% } %>><% } %><% if (databaseType === 'mongodb') { %>MongoRepository<User, String><% } %><% if (databaseType === 'couchbase') { %>N1qlCouchbaseRepository<User, String><% } %> {
-    <%_ if (cacheManagerIsAvailable) { _%>
+    <%_ if (cacheManagerIsAvailable === true) { _%>
 
     String USERS_BY_LOGIN_CACHE = "usersByLogin";
 

